### PR TITLE
Yeoman generator for new feature

### PR DESCRIPTION
**Pull request by jitinmaher**
# Title

yeoman generator support for new feature

## Description of what's changing

The changes include adding a generator for scaffolding a new feature similar to `HMRC`.

To scaffold, below are the steps:
- `yo fraudpreventiondata`
- Enter the component name in the prompt
- A scaffolded version of `generator-fraudulentlogic/generators/templates/helloworld` will be created in `src/js` 

## What else might be impacted?
The changes adds a new npm module(yeoman) to the project

## Which issue does this PR relate to?

https://github.com/intuit/user-data-for-fraud-prevention/issues/16

## Checklist

[ ] Tests are written and maintain or improve code coverage
[ ] I've tested this in my application using `yarn link` (if applicable)
[ ] You consent to and are confident this change can be released with no regression

### CODE REVIEW

**Comment by jitinmaher** [comment]
@skodamarthi  @reubenae can you please help with the review? Thanks!

File: generator-fraudulentlogic/generators/templates/hmrc/mtdFraudPrevention.js
Comment (reubenae) [inline generator-fraudulentlogic/generators/templates/hmrc/mtdFraudPrevention.js:1]: I would rather this be "hello world" rather than a fully fledged package

File: generator-fraudulentlogic/generators/templates/hmrc/README.md
Comment (reubenae) [inline generator-fraudulentlogic/generators/templates/hmrc/README.md:1]: I'd probably just give a skeleton outline for the readme: title, description, usage maybe?

File: generator-fraudulentlogic/generators/index.js
Comment (reubenae) [inline generator-fraudulentlogic/generators/index.js:5]: whats with the title "fraudulentlogic"?

File: package.json
Comment (reubenae) [inline package.json:54]: Shouldn't this be in devDependencies? Consumers of this package don't need this to ship to customers :)

File: generator-fraudulentlogic/package.json
Comment (reubenae) [inline generator-fraudulentlogic/package.json:10]: Do we need this package.json at all?

**Review by reubenae — CHANGES_REQUESTED** [review]
Thanks for your contribution! This is a good start. Some changes requested and some questions.

File: generator-fraudulentlogic/generators/index.js
Comment (jitinmaher) [reply generator-fraudulentlogic/generators/index.js:5]: @reubenae can you suggest some better name for this generator, I couldn't come up with a good name

File: package.json
Comment (jitinmaher) [reply package.json:54]: yes agreed! Misplaced it.

File: generator-fraudulentlogic/package.json
Comment (jitinmaher) [reply generator-fraudulentlogic/package.json:10]: I was going through the docs and it says we should have this under the generators folder. https://yeoman.io/authoring/

File: generator-fraudulentlogic/generators/index.js
Comment (skodamarthi) [reply generator-fraudulentlogic/generators/index.js:5]: @reubenae @jitinmaher How does generator-fraudpreventiondata sound?

**Comment by skodamarthi** [comment]
@jitinmaher Looks fine to me. But, will wait for @reubenae to approve the changes before merging as he has requested changes on the PR originally. He is currently out on vacation and will be back by this Friday. Thanks for your patience!

**Comment by reubenae** [comment]
Thanks for your contribution @jitinmaher - am going to pull your code locally and play around with it. Will update with any comments

File: generator-fraudpreventiondata/generators/templates/hmrc/README.md
Comment (reubenae) [inline generator-fraudpreventiondata/generators/templates/hmrc/README.md:1]: I would probably just call this a `helloworld` template. Since that is what it is. So path `generators/templates/helloworld`

File: generator-fraudpreventiondata/generators/index.js
Comment (reubenae) [inline generator-fraudpreventiondata/generators/index.js:3]: I wasn't able to test this, because I wasn't sure how to run the generator. Update the README contributing section with some instructions.

**Review by reubenae — CHANGES_REQUESTED** [review]
Looks like we're mostly there. Two comments for you to address, then I think we can merge (after a test)

File: generator-fraudpreventiondata/generators/index.js
Comment (jitinmaher) [reply generator-fraudpreventiondata/generators/index.js:3]: 👍

File: generator-fraudpreventiondata/generators/templates/hmrc/README.md
Comment (jitinmaher) [reply generator-fraudpreventiondata/generators/templates/hmrc/README.md]: 👍

**Comment by jitinmaher** [comment]
@reubenae Updated with review comments. Let me know if it looks good.

**Comment by jitinmaher** [comment]
@reubenae @skodamarthi any updates on this?

**Comment by reubenae** [comment]
@jitinmaher sorry for the slow response. Can you remove the package-lock.json file? yarn and npm are exclusive package managers. So you may need to update your yarn.lock file too (just run `yarn` from the root to do this). If you have trouble with this you can add me as a contributor to your fork and I can push a fix from my local

**Comment by reubenae** [comment]
Following your steps in the readme, this still doesn't work for me. I think your guide needs a bit more work, e.g. what commands do I need to run to make sure yeoman is installed? How do I install the new generator you've created? What directory should I be in to run it? Here is the error I get:

image

**Comment by reubenae** [comment]
Going to close this PR for now as it seems to have gone stale, please feel free to re-open it addressing the above comments

**Comment by jitinmaher** [comment]
Hey @reubenae @skodamarthi , is this issue still relevant? Can this be reopened? We can do some quick follow ups on the last point that we dropped off from the conversation. Let me know. Thanks!

**Pull request closed without merging**
